src/Network/Network.Test/ScenarioTests/NatGatewayTests.ps1

```diff
@@ -0,0 +1,293 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# 
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# 
+# 
+# Warning: This code was generated by a tool.
+# 
+# Changes to this file may cause incorrect behavior and will be lost if the
+# code is regenerated.
+# 
+# For documentation on code generator please visit
+#   https://aka.ms/nrp-code-generation
+# Please contact wanrpdev@microsoft.com if you need to make changes to this file.
+# </auto-generated>
+
+function Check-CmdletReturnType
+{
+    param($cmdletName, $cmdletReturn)
+
+    $cmdletData = Get-Command $cmdletName;
+    Assert-NotNull $cmdletData;
+    [array]$cmdletReturnTypes = $cmdletData.OutputType.Name | Foreach-Object { return ($_ -replace "Microsoft.Azure.Commands.Network.Models.","") };
+    [array]$cmdletReturnTypes = $cmdletReturnTypes | Foreach-Object { return ($_ -replace "System.","") };
+    $realReturnType = $cmdletReturn.GetType().Name -replace "Microsoft.Azure.Commands.Network.Models.","";
+    return $cmdletReturnTypes -contains $realReturnType;
+}
+
+<#
+.SYNOPSIS
+Test creating new NatGateway using minimal set of parameters
+#>
+function Test-NatGatewayCRUDMinimalParameters
+{
+    # Setup
+    $rgname = Get-ResourceGroupName;
+    $rglocation = Get-ProviderLocation ResourceManagement;
+    $rname = Get-ResourceName;
+    $location = "eastus2";
+
+    try
+    {
+        $resourceGroup = New-AzResourceGroup -Name $rgname -Location $rglocation;
+
+        # Create NatGateway
+        $vNatGateway = New-AzNatGateway -ResourceGroupName $rgname -Name $rname -Location $location -Sku Standard;
+        Assert-NotNull $vNatGateway;
+        Assert-True { Check-CmdletReturnType "New-AzNatGateway" $vNatGateway };
+        Assert-AreEqual $rname $vNatGateway.Name;
+
+        # Get NatGateway
+        $vNatGateway = Get-AzNatGateway -ResourceGroupName $rgname -Name $rname;
+        Assert-NotNull $vNatGateway;
+        Assert-True { Check-CmdletReturnType "Get-AzNatGateway" $vNatGateway };
+        Assert-AreEqual $rname $vNatGateway.Name;
+
+        # Get all NatGateways in resource group
+        $listNatGateway = Get-AzNatGateway -ResourceGroupName $rgname;
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription
+        $listNatGateway = Get-AzNatGateway;
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription wildcard for resource group
+        $listNatGateway = Get-AzNatGateway -ResourceGroupName "*";
+        Assert-NotNull ($listNatGateway | Where-Object { $_.ResourceGroupName -eq $rgname -and $_.Name -eq $rname });
+
+        # Get all NatGateways in subscription wildcard for name
+        $listNatGateway = Get-AzNatGateway -Name "*";
```

Comment:
Parameter set would be resolved to `GetByNameParameterSet` and it requires `ResourceGroupName` specified. So please remove calls of this kind as they would fail.